### DISCUSSION TITLE
How add a custom layout and switch different layout for the content?

### DISCUSSION THREAD

**Discussion by gaojianzhuang** [OrchardCMS/OrchardCore#16878] [Q&A]
Sometimes, we need to create different layouts for the different content.
For now, the theme only has one "Layout.liquid" file, if I create a new layout file "Layout-LeftBar.liquid", how to use it?
![image](https://github.com/user-attachments/assets/9b54f5f6-5ee1-46e1-956f-b030020153dc)

**Comment by Piedone** [comment]
I don't think we have built-in alternates for the Layout shape, but you don't need those for such a scenario anyway. Check out the different Content alternates: https://docs.orchardcore.net/en/latest/reference/modules/Templates/#content-templates

**Reply by gaojianzhuang** [reply]
Thanks Piedone. I have tried to define a different content template, but it only supports the content level. Sometimes, we need the site-level template. How to define it? Thanks.

**Reply by Piedone** [reply]
You could add your own [alternate](https://docs.orchardcore.net/en/latest/reference/modules/Themes/#alternates), which is easy enough with a bit of coding, see e.g. [`WidgetAliasShapeTableProvider`](https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore.Modules/OrchardCore.Alias/Services/WidgetAliasShapeTableProvider.cs).

**Comment by hishamco** [comment]
Why the `Layout-LeftBar.liquid` is not a zone?

Regarding your question, I have never seen such an alternate. Please elaborate of what you are trying to do

**Reply by gaojianzhuang** [reply]
![image](https://github.com/user-attachments/assets/f83c586f-38ba-484b-84e1-b39aab0ab2b8)
For example, we need two layouts in one site. We need to switch between different layouts for different content.
My question is how to define different layouts with orchard core. Could you please introduce the solution? 
Thanks so much. :)

**Reply by hishamco** [reply]
What if you switch between the layouts after checking the content item by slug?

**Reply by gaojianzhuang** [reply]
I think it should be the content type level not content item level. We can config the mapping for the content types with the layouts.

**Comment by ns8482e** [comment]
You can definitely use `Layout-LeftBar.liquid` and `Layout.liquid` same time without switching theme. There is built in property that adds alternates to your `Layout`.

in C# - cshtml use `ViewLayout = 'Layout__LeftBar'` and in liquid use `{% layout "Layout__LeftBar" %}`
